### PR TITLE
[Plot] Support domain/range switching

### PR DESCRIPTION
**Pull request by VWoeltjen**
Opening up pull request early for visibility, early feedback, to-do tracking.

Addresses #218 by responding to user selection of domains/ranges.

To-do:
- [x] Requery on domain/range changes
- [x] Change data shown for different ranges
- [x] Change data shown for different domains
- [x] Verify reasonable behavior with time conductor
- [x] Update tests

### CODE REVIEW

**Comment by VWoeltjen** [comment]
Summary of changes:

* Watch for changes to user selection of domain/range and requery when this occurs
  * Also, update timestamp formatting to reflect changes in domain format
* Allow specification of domain/range keys when requesting latest data from TelemetrySubscription (instead of always returning the default domain/range values)

### Author Checklist

1. Changes address original issue? Y
2. Unit tests included and/or updated with changes? Y
3. Command line build passes? Y
4. Expect to pass code review? Y

File: platform/telemetry/src/TelemetrySubscription.js [hunk 1]
```diff
@@ -287,9 +287,12 @@ define(
          * @param {DomainObject} domainObject the object of interest
          * @returns the most recent domain value observed
          */
-        TelemetrySubscription.prototype.getDomainValue = function (domainObject) {
-            var id = domainObject.getId();
-            return (this.latestValues[id] || {}).domain;
+        TelemetrySubscription.prototype.getDomainValue = function (domainObject, key) {
```
Comment (akhenry) [inline platform/telemetry/src/TelemetrySubscription.js]: jsdoc needs updating for new parameter

File: platform/telemetry/src/TelemetrySubscription.js [hunk 2]
```diff
@@ -304,9 +307,12 @@ define(
          * @param {DomainObject} domainObject the object of interest
          * @returns the most recent range value observed
          */
-        TelemetrySubscription.prototype.getRangeValue = function (domainObject) {
-            var id = domainObject.getId();
-            return (this.latestValues[id] || {}).range;
+        TelemetrySubscription.prototype.getRangeValue = function (domainObject, key) {
```
Comment (akhenry) [inline platform/telemetry/src/TelemetrySubscription.js]: Ditto

**Comment by akhenry** [comment]
Some minor additional jsdoc required. Otherwise looks fine to me.

**Comment by VWoeltjen** [comment]
Good catch - added missing JSDoc, reassigning for review/merge

**Comment by akhenry** [comment]
### Reviewer Checklist

Changes appear to address issue? __Y__
Appropriate unit tests included? __Y__
Code style and in-line documentation are appropriate? __Y__
Commit messages meet standards? __Y__

**Pull request merged**